app-backend/db/src/main/scala/com/azavea/rf/database/ObjectPermissions.scala

```diff
@@ -0,0 +1,127 @@
+package com.azavea.rf.database
+
+import com.azavea.rf.datamodel._
+import com.azavea.rf.database.Implicits._
+import doobie._
+import doobie.implicits._
+import doobie.postgres._
+import doobie.postgres.implicits._
+import cats._
+import cats.data._
+import cats.effect.IO
+import cats.implicits._
+import java.util.UUID
+
+trait ObjectPermissions {
+  def tableName: String
+
+  def isValidObject(id: UUID): ConnectionIO[Boolean] = (tableName match {
+    case "projects"    => ProjectDao
+    case "scenes"      => SceneDao
+    case "datasources" => DatasourceDao
+    case "shapes"      => ShapeDao
+    case "workspaces" | "templates" | "analyses" =>
+      throw new Exception(s"${tableName} not yet supported")
+  }).query.filter(id).exists
+
+  def isValidPermission(acr: ObjectAccessControlRule): ConnectionIO[Boolean] = (acr.subjectType, acr.subjectId) match {
+    case (SubjectType.All, _) => true.pure[ConnectionIO]
+    case (SubjectType.Platform, Some(sid)) =>
```

Comment:
I think making all of these `sid` calls be `subjectId` would be good -- there's a little bit of mental overhead coming back to this and trying to remember what `s` means. It's easy enough to go back and figure it out here -- but I don't think there's much gained by the abbreviation. Kind of a nitpick, but I think it improves readability.